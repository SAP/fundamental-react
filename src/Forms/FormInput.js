import classnamesBind from 'classnames/bind';
import { FORM_MESSAGE_TYPES } from '../utils/constants';
import FormValidationOverlay from './_FormValidationOverlay';
import PropTypes from 'prop-types';
import React from 'react';
import withStyles from '../utils/withStyles';
import styles from 'fundamental-styles/dist/input.css';

const classnames = classnamesBind.bind(styles);
/** A **FormInput** is used to collect data from the user. When a field is required,
the \`required\` property will include an asterisk (*). */
const FormInput = React.forwardRef(({
    className,
    cssNamespace,
    compact,
    disabled,
    name,
    placeholder,
    readOnly,
    type,
    value,
    validationOverlayProps,
    validationState,
    ...props
}, ref) => {

    const formInputClasses = classnames(
        `${cssNamespace}-input`,
        {
            [`${cssNamespace}-input--compact`]: !!compact,
            [`is-${validationState?.state}`]: validationState?.state
        },
        className
    );

    const formInput = (
        <input
            {...props}
            className={formInputClasses}
            disabled={disabled}
            name={name}
            placeholder={placeholder}
            readOnly={readOnly}
            ref={ref}
            type={type}
            value={value} />
    );

    return validationState ? (
        <FormValidationOverlay
            {...validationOverlayProps}
            control={formInput}
            validationState={validationState} />
    ) : formInput;
});

FormInput.displayName = 'FormInput';

FormInput.propTypes = {
    /** CSS class(es) to add to the element */
    className: PropTypes.string,
    /** Set to **true** to enable compact mode */
    compact: PropTypes.bool,
    /** Set to **true** to mark component as disabled and make it non-interactive */
    disabled: PropTypes.bool,
    /** Value for the `name` attribute on the input */
    name: PropTypes.string,
    /** Localized placeholder text of the input */
    placeholder: PropTypes.string,
    /** Set to **true** to mark component as readonly */
    readOnly: PropTypes.bool,
    /** Value for the `type` attribute on the input */
    type: PropTypes.string,
    /** Additional props to be spread to the ValidationOverlay */
    validationOverlayProps: PropTypes.shape({
        /** Additional classes to apply to validation popover's outermost `<div>` element  */
        className: PropTypes.string,
        /** Additional props to be spread to the ValdiationOverlay's FormMessage component */
        formMessageProps: PropTypes.object,
        /** Additional classes to apply to validation popover's popper child `<div>` wrapping the provided children  */
        innerRefClassName: PropTypes.string,
        /** Additional classes to apply to validation popover's popper `<div>` element  */
        popperClassName: PropTypes.string,
        /** CSS class(es) to add to the ValidationOverlay's reference `<div>` element */
        referenceClassName: PropTypes.string,
        /** Additional props to be spread to the popover's outermost `<div>` element */
        wrapperProps: PropTypes.object
    }),
    /** An object identifying a validation message.  The object will include properties for `state` and `text`; _e.g._, \`{ state: \'warning\', text: \'This is your last warning\' }\` */
    validationState: PropTypes.shape({
        /** State of validation: 'error', 'warning', 'information', 'success' */
        state: PropTypes.oneOf(FORM_MESSAGE_TYPES),
        /** Text of the validation message */
        text: PropTypes.string
    }),
    /** Value for the `value` attribute on the input */
    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
};

FormInput.defaultProps = {
    type: 'text'
};

export default withStyles(FormInput);
