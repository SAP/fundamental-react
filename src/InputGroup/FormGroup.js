import PropTypes from 'prop-types';
import React from 'react';
import withStyles from '../utils/WithStyles/WithStyles';

const FormGroup = React.forwardRef(({ children, disableStyles, ...props }, ref) => {
    return (<div {...props} className='fd-form-group'
        ref={ref}>{children}</div>);
});

FormGroup.displayName = 'FormGroup';

FormGroup.propTypes = {
    children: PropTypes.node,
    customStyles: PropTypes.object,
    disableStyles: PropTypes.bool
};

export { FormGroup as __FormGroup };

export default withStyles(FormGroup, { cssFile: 'form-group' });
