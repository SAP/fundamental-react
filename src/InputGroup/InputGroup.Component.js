import path from 'path';
import React from 'react';
import { Button, FormGroup, FormInput, FormItem, FormLabel, Icon, InputGroup } from '../';
import { ComponentPage, Example } from '../_playground';

export const InputGroupComponent = () => {
    return (
        <ComponentPage
            description={`The **Input Group** includes form inputs with add-ons that allow the user to
                better understand the information being entered.`}
            sourceModulePath={path.join(__dirname, './InputGroup')}
            title='Input Group'>

            <Example
                description={`The Input with text add-on component is typically used to specify the
                    type of the data being entered, such as currency or unit of measure.
                    This add-on can be placed at the left or right of the input element.`}
                title='Text add-on'>
                <div>
                    <FormGroup>
                        <FormLabel>Left Aligned Text Addon</FormLabel>
                        <FormItem>
                            <InputGroup>
                                <InputGroup.Addon>$</InputGroup.Addon>
                                <FormInput
                                    placeholder='Type text here' />
                            </InputGroup>
                        </FormItem>
                    </FormGroup>
                    <br />
                    <FormGroup>
                        <FormLabel>Right Aligned Text Addon</FormLabel>
                        <FormItem>
                            <InputGroup>
                                <FormInput placeholder='Type text here' />
                                <InputGroup.Addon>€</InputGroup.Addon>
                            </InputGroup>
                        </FormItem>
                    </FormGroup>
                    <br />
                    <p>Compact mode</p>
                    <FormGroup>
                        <FormLabel>Left Aligned Text Addon</FormLabel>
                        <FormItem>
                            <InputGroup compact>
                                <InputGroup.Addon>$</InputGroup.Addon>
                                <FormInput placeholder='Type text here' />
                            </InputGroup>
                        </FormItem>
                    </FormGroup>
                    <br />
                    <FormGroup>
                        <FormLabel>Right Aligned Text Addon</FormLabel>
                        <FormItem>
                            <InputGroup compact>
                                <FormInput
                                    placeholder='Type text here' />
                                <InputGroup.Addon>€</InputGroup.Addon>
                            </InputGroup>
                        </FormItem>
                    </FormGroup>
                </div>
            </Example>

            <Example

                description='The consumer can add an Icon as a child of InputGroup.Addon.'
                title='Input with icons'>
                <div>
                    <FormGroup>
                        <FormLabel>Input with icon on the left</FormLabel>
                        <FormItem>
                            <InputGroup>
                                <InputGroup.Addon>
                                    <Icon glyph='globe' />
                                </InputGroup.Addon>
                                <FormInput placeholder='Type text here' />
                            </InputGroup>
                        </FormItem>
                    </FormGroup>

                    <br />
                    <p>Compact mode</p>
                    <FormGroup>
                        <FormLabel>Input with icon on the left</FormLabel>
                        <FormItem>
                            <InputGroup compact>
                                <InputGroup.Addon>
                                    <Icon glyph='globe' />
                                </InputGroup.Addon>
                                <FormInput placeholder='Type text here' />
                            </InputGroup>
                        </FormItem>
                    </FormGroup>
                    <br />
                    <br />
                    <FormGroup>
                        <FormLabel>Input with icon on the right</FormLabel>
                        <FormItem>
                            <InputGroup>
                                <FormInput placeholder='Type text here' />
                                <InputGroup.Addon>
                                    <Icon glyph='hide' />
                                </InputGroup.Addon>
                            </InputGroup>
                        </FormItem>
                    </FormGroup>
                    <br />
                    <p>Compact mode</p>
                    <FormGroup>
                        <FormLabel>Input with icon on the right</FormLabel>
                        <FormItem>
                            <InputGroup compact>
                                <FormInput placeholder='Type text here' />
                                <InputGroup.Addon>
                                    <Icon glyph='hide' />
                                </InputGroup.Addon>
                            </InputGroup>
                        </FormItem>
                    </FormGroup>
                </div>
            </Example>

            <Example
                description={`The InputGroup.Addon isButton supports buttons by providing the correct styling. Buttons can be shown with a text
                label or icon.`}
                title='Input with Buttons'>
                <div>
                    <FormGroup>
                        <FormLabel>Input text with action</FormLabel>
                        <FormItem>
                            <InputGroup >
                                <FormInput placeholder='Type text here' />
                                <InputGroup.Addon isButton>
                                    <Button option='transparent'>Button</Button>
                                </InputGroup.Addon>
                            </InputGroup>
                        </FormItem>
                    </FormGroup>
                    <br />
                    <p>Compact mode</p>
                    <FormGroup>
                        <FormLabel>Input text with action</FormLabel>
                        <FormItem>
                            <InputGroup compact>
                                <FormInput placeholder='Type text here' />
                                <InputGroup.Addon isButton>
                                    <Button option='transparent'>Button</Button>
                                </InputGroup.Addon>
                            </InputGroup>
                        </FormItem>
                    </FormGroup>
                    <br />
                    <br />
                    <FormGroup>
                        <FormLabel>Input with icon text action</FormLabel>
                        <FormItem>
                            <InputGroup>
                                <FormInput />
                                <InputGroup.Addon isButton>
                                    <Button
                                        glyph='navigation-down-arrow'
                                        option='transparent' />
                                </InputGroup.Addon>
                            </InputGroup>
                        </FormItem>
                    </FormGroup>
                    <br />
                    <p>Compact mode</p>
                    <FormGroup>
                        <FormLabel>Input with icon text action</FormLabel>
                        <FormItem>
                            <InputGroup compact>
                                <FormInput placeholder='Type text here' />
                                <InputGroup.Addon isButton>
                                    <Button
                                        glyph='navigation-down-arrow'
                                        option='transparent' />
                                </InputGroup.Addon>
                            </InputGroup>
                        </FormItem>
                    </FormGroup>
                </div>
            </Example>
        </ComponentPage>
    );
};
