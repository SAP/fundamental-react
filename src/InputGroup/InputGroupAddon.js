import Button from '../Button/Button';
import classnames from 'classnames';
import CustomPropTypes from '../utils/CustomPropTypes/CustomPropTypes';
import Icon from '../Icon/Icon';
import PropTypes from 'prop-types';
import React from 'react';
import withStyles from '../utils/WithStyles/WithStyles';

const InputGroupAddon = ({
    addon,
    compact,
    actions,
    glyph,
    addonClassNames,
    children,
    className,
    disableStyles,
    inputType,
    numberDownCallback,
    numberUpCallback,
    localizedText,
    numberDownButtonProps,
    numberUpButtonProps,
    onClick,
    ...otherProps
}) => {

    let returnObject = addon;

    if (actions) {
        returnObject = children;
    } else if (glyph) {
        returnObject = (
            <Icon
                disableStyles={disableStyles}
                glyph={glyph}
                role='presentation' />
        );
    }

    const addonClasses = classnames(
        addonClassNames,
        'fd-input-group__addon',
        [{ 'fd-input-group__addon--button': !!actions || inputType === 'number' }],
        [{ 'fd-input-group__addon--button--compact': compact }],
    );

    if (inputType === 'number') {
        return (
            <span {...otherProps} className={addonClasses}>
                <Button
                    {...numberUpButtonProps}
                    aria-label={localizedText.up}
                    className='fd-button--half fd-input-group__button'
                    compact={compact}
                    disableStyles={disableStyles}
                    glyph='slim-arrow-up'
                    onClick={numberUpCallback}
                    option='light' />
                <Button
                    {...numberDownButtonProps}
                    aria-label={localizedText.down}
                    className='fd-button--half fd-input-group__button'
                    compact={compact}
                    disableStyles={disableStyles}
                    glyph='slim-arrow-down'
                    onClick={numberDownCallback}
                    option='light' />
            </span>
        );
    } else {
        return (
            <span className={addonClasses} {...otherProps} >
                {returnObject}
            </span>
        );
    }
};


InputGroupAddon.displayName = 'InputGroupAddon';

InputGroupAddon.propTypes = {
    addon: PropTypes.string,
    addonClassNames: PropTypes.string,
    children: PropTypes.node,
    className: PropTypes.string,
    customStyles: PropTypes.object,
    disableStyles: PropTypes.bool,
    glyph: PropTypes.string,
    localizedText: CustomPropTypes.i18n({
        clear: PropTypes.string,
        down: PropTypes.string,
        up: PropTypes.string
    }),
    numberDownButtonProps: PropTypes.object,
    numberUpButtonProps: PropTypes.object,
    onClick: PropTypes.func
};

InputGroupAddon.defaultProps = {
    localizedText: {
        clear: 'Clear',
        down: 'Step down',
        up: 'Step up'
    }
};

InputGroupAddon.propDescriptions = {
    addon: 'The value of the add-on.',
    addonClassNames: 'CSS class(es) to add to the addon element.',
    localizedTextShape: {
        clear: 'Value for aria-label on the clear <button> element.',
        down: 'Value for aria-label on the down <button> element.',
        up: 'Value for aria-label on the up <button> element.'
    },
    numberDownButtonProps: 'Additional props to be spread to the down `<button>` element (for inputType=\'number\').',
    numberUpButtonProps: 'Additional props to be spread to the up `<button>` element (for inputType=\'number\').'
};

export { InputGroupAddon as __InputGroupAddon };

export default withStyles(InputGroupAddon, { cssFile: 'input-group', fonts: true });
