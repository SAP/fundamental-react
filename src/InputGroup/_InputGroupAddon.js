import classnamesBind from 'classnames/bind';
import PropTypes from 'prop-types';
import React from 'react';
import withStyles from '../utils/withStyles';
import styles from 'fundamental-styles/dist/input-group.css';

const classnames = classnamesBind.bind(styles);

const InputGroupAddon = ({
    children,
    className,
    compact,
    cssNamespace,
    isButton,
    ...otherProps
}) => {

    const addonClasses = classnames(
        className,
        `${cssNamespace}-input-group__addon`,
        {
            [`${cssNamespace}-input-group__addon--button`]: isButton,
            [`${cssNamespace}-input-group__addon--compact`]: compact
        }
    );

    return (
        <span className={addonClasses} {...otherProps} >
            {React.Children.toArray(children).map(child => {
                if (!isButton) {
                    return child;
                }

                return React.cloneElement(child, {
                    compact,
                    className: classnames(
                        child.props.className,
                        `${cssNamespace}-input-group__button`,
                    )
                });
            })}
        </span>
    );
};

InputGroupAddon.displayName = 'InputGroup.Addon';

InputGroupAddon.propTypes = {
    /** Node(s) to render within the component */
    children: PropTypes.node,
    /** CSS class(es) to add to the element */
    className: PropTypes.string,
    /** Internal use only */
    compact: PropTypes.bool,
    /** Set to **true** if add-on is a button */
    isButton: PropTypes.bool
};

export default withStyles(InputGroupAddon);
