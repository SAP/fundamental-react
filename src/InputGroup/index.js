export { default as FormGroup } from './FormGroup.js';
export { default as InputGroup } from './InputGroup.js';
