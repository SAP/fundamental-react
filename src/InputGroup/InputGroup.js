import classnamesBind from 'classnames/bind';
import { FORM_MESSAGE_TYPES } from '../utils/constants';
import FormValidationOverlay from '../Forms/_FormValidationOverlay';
import InputGroupAddon from './_InputGroupAddon';
import PropTypes from 'prop-types';
import withStyles from '../utils/withStyles';
import React, { Component } from 'react';
import styles from 'fundamental-styles/dist/input-group.css';

const classnames = classnamesBind.bind(styles);
const isUsingCssModules = styles && Object.keys(styles).length > 0;

/** An **InputGroup** includes form inputs with add-ons that allow the user to
better understand the information being entered. */
class InputGroup extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        const {
            children,
            className,
            compact,
            cssNamespace,
            disabled,
            validationOverlayProps,
            validationState,
            ...props
        } = this.props;

        const inputGroupClasses = classnames(
            className,
            `${cssNamespace}-input-group`,
            {
                'is-disabled': disabled,
                [`is-${validationState?.state}`]: validationState?.state
            },
        );

        const getClassName = (child) => classnames(
            {
                [`${cssNamespace}-input-group__input`]: !child.props.className?.includes(`${cssNamespace}-tokenizer`),
                [`${cssNamespace}-textarea`]: child.type?.displayName === 'FormTextarea' && isUsingCssModules
            },
            child.props.className
        );

        const inputGroup = (
            <div
                {...props}
                className={inputGroupClasses}>
                {React.Children.toArray(children).map(child => {
                    if (child?.type?.displayName === InputGroupAddon.displayName) {
                        return React.cloneElement(child, {
                            compact,
                            disabled
                        });
                    }
                    return React.cloneElement(child, {
                        compact,
                        disabled,
                        className: getClassName(child),
                        validationState: child.props.validationState && null // remove duplicate validation state from child component
                    });
                })}
            </div>
        );

        return (
            <FormValidationOverlay
                {...validationOverlayProps}
                control={inputGroup}
                validationState={validationState} />
        );
    }
}

InputGroup.Addon = InputGroupAddon;

InputGroup.displayName = 'InputGroup';

InputGroup.propTypes = {
    /** Node(s) to render within the component */
    children: PropTypes.node,
    /** CSS class(es) to add to the element */
    className: PropTypes.string,
    /** Set to **true** to enable compact mode */
    compact: PropTypes.bool,
    /** Set to **true** to mark component as disabled and make it non-interactive */
    disabled: PropTypes.bool,
    /** Additional props to be spread to the ValidationOverlay */
    validationOverlayProps: PropTypes.shape({
        /** Additional classes to apply to validation popover's outermost `<div>` element  */
        className: PropTypes.string,
        /** Additional props to be spread to the ValdiationOverlay's FormMessage component */
        formMessageProps: PropTypes.object,
        /** Additional classes to apply to validation popover's popper child `<div>` wrapping the provided children  */
        innerRefClassName: PropTypes.string,
        /** Additional classes to apply to validation popover's popper `<div>` element  */
        popperClassName: PropTypes.string,
        /** CSS class(es) to add to the ValidationOverlay's reference `<div>` element */
        referenceClassName: PropTypes.string,
        /** Additional props to be spread to the popover's outermost `<div>` element */
        wrapperProps: PropTypes.object
    }),
    /** An object identifying a validation message.  The object will include properties for `state` and `text`; _e.g._, \`{ state: \'warning\', text: \'This is your last warning\' }\` */
    validationState: PropTypes.shape({
        /** State of validation: 'error', 'warning', 'information', 'success' */
        state: PropTypes.oneOf(FORM_MESSAGE_TYPES),
        /** Text of the validation message */
        text: PropTypes.string
    })
};

export default withStyles(InputGroup);
