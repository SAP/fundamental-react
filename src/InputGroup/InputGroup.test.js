import { act } from 'react-dom/test-utils';
import FormInput from '../Forms/FormInput';
import InputGroup from './InputGroup';
import { mount } from 'enzyme';
import React from 'react';

describe('<InputGroup />', () => {
    let setup = (props) => {
        return mount(
            <InputGroup {...props}>
                <InputGroup.Addon>$</InputGroup.Addon>
                <FormInput />
            </InputGroup>);
    };

    describe('Default Rendering', () => {
        let element = setup();

        it('should render', () => {
            expect(
                element
            ).toBeDefined();
        });

        it('should render a div tag by default', () => {
            expect(
                element.getDOMNode().tagName
            ).toBe('DIV');
        });

        it('should have a default class of "fd-input-group"', () => {
            expect(
                element.find('.fd-popover__control').childAt(0).getDOMNode().className
            ).toContain('fd-input-group');
        });
    });

    describe('Rendering with Props', () => {
        it('should allow data attribute to be passed to the element', () => {
            const element = setup({
                'data-sample': 'Sample'
            });

            expect(
                element.find('.fd-popover__control').childAt(0).getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });

        it('should pass correct classnames if child.type is an input', () => {
            let element = setup();

            expect(
                element.find('.fd-input').getDOMNode().className
            ).toContain('fd-input-group__input');
            expect(
                element.getDOMNode().children[0].className
            ).not.toContain('fd-input-group__input');
        });

        describe('validationOverlayProps', () => {
            afterEach(() => {
                document.body.innerHTML = '';
            });

            const getFormMessage = () => document.body.querySelector('.fd-popover__popper > div > .fd-form-message');


            test('should allow spreading className to ValidationOverlay popover', () => {
                const wrapper = setup({
                    validationState: { state: 'error', text: 'Test validation state' },
                    validationOverlayProps: { className: 'wonderful-styles' }
                });

                expect(
                    wrapper.find('.fd-popover').getDOMNode().classList
                ).toContain('wonderful-styles');
            });

            test('should allow spreading className to ValidationOverlay innerRef div', () => {
                const wrapper = setup({
                    validationState: { state: 'error', text: 'Test validation state' },
                    validationOverlayProps: { innerRefClassName: 'wonderful-styles', show: true }
                });

                expect(
                    wrapper.find('.fd-popover__innerRef').getDOMNode().classList
                ).toContain('wonderful-styles');
            });

            test('should allow spreading className to ValidationOverlay reference div', () => {
                const wrapper = setup({
                    validationState: { state: 'error', text: 'Test validation state' },
                    validationOverlayProps: { referenceClassName: 'wonderful-styles' }
                });

                expect(
                    wrapper.find('.fd-popover__control').getDOMNode().classList
                ).toContain('wonderful-styles');
            });

            test('should spread formMessageProps to ValidationOverlay\'s FormMessage', async() => {
                await act(async() => {
                    setup({
                        validationState: {
                            state: 'error',
                            text: 'Test validation state'
                        },
                        validationOverlayProps: {
                            formMessageProps: { 'data-sample': 'Sample', className: 'wonderful-styles' },
                            show: true
                        }
                    });
                });

                const messageNode = getFormMessage();

                expect(
                    messageNode.attributes['data-sample'].value
                ).toBe('Sample');

                expect(
                    messageNode.classList
                ).toContain('wonderful-styles');
            });

            test('should spread props to Validation overlay wrapper div', async() => {
                const wrapper = setup({
                    validationState: { state: 'error', text: 'Test validation state' },
                    validationOverlayProps: { wrapperProps: { 'data-sample': 'Sample' }, show: true }
                });

                expect(
                    wrapper.find('.fd-popover').getDOMNode().attributes['data-sample'].value
                ).toBe('Sample');
            });

            test('should set class on the Validation Overlay Popper', async() => {
                await act(async() => {
                    setup({
                        validationState: { state: 'error', text: 'Test validation state' },
                        validationOverlayProps: { popperClassName: 'wonderful-styles', show: true }
                    });
                });

                expect(
                    document.body.querySelector('.fd-popover__popper').classList
                ).toContain('wonderful-styles');
            });
        });
    });
});
