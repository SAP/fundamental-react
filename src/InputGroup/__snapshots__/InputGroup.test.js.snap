// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputGroup /> create input group items 1`] = `
<div
  className="fd-input-group fd-input-group--before"
>
  <span
    className="fd-input-group__addon fd-input-group__addon--before"
  >
    $
  </span>
  <input
    className=""
    onChange={[Function]}
    type="text"
    value="1234567890"
  />
</div>
`;

exports[`<InputGroup /> create input group items 2`] = `
<div
  className="fd-input-group fd-input-group--after"
>
  <input
    className=""
    onChange={[Function]}
    type="text"
    value="1234567890"
  />
  <span
    className="fd-input-group__addon fd-input-group__addon--after"
  >
    €
  </span>
</div>
`;

exports[`<InputGroup /> create input group items 3`] = `
<div
  className="fd-input-group fd-input-group--before fd-input-group--compact"
>
  <span
    className="fd-input-group__addon fd-input-group__addon--before"
  >
    $
  </span>
  <input
    className="fd-input fd-input--compact"
    onChange={[Function]}
    type="text"
    value="1234567890"
  />
</div>
`;

exports[`<InputGroup /> create input group items 4`] = `
<div
  className="fd-input-group fd-input-group--after fd-input-group--compact"
>
  <input
    className="fd-input fd-input--compact"
    onChange={[Function]}
    type="text"
    value="1234567890"
  />
  <span
    className="fd-input-group__addon fd-input-group__addon--after"
  >
    €
  </span>
</div>
`;

exports[`<InputGroup /> create input group items 5`] = `
<div
  className="fd-input-group fd-input-group--after"
>
  <input
    className=""
    onChange={[Function]}
    type="number"
    value={100}
  />
  <span
    className="fd-input-group__addon fd-input-group__addon--button fd-input-group__addon--after"
  >
    <button
      aria-label="Step up"
      className="fd-input-group__button fd-input-group__button--step-up sap-icon--slim-arrow-up"
      onClick={[Function]}
    />
    <button
      aria-label="Step down"
      className="fd-input-group__button fd-input-group__button--step-down sap-icon--slim-arrow-down"
      onClick={[Function]}
    />
  </span>
</div>
`;

exports[`<InputGroup /> create input group items 6`] = `
<div
  className="fd-input-group fd-input-group--after fd-input-group--compact"
>
  <input
    className="fd-input fd-input--compact"
    onChange={[Function]}
    type="number"
    value={100}
  />
  <span
    className="fd-input-group__addon fd-input-group__addon--button fd-input-group__addon--after"
  >
    <button
      aria-label="Step up"
      className="fd-input-group__button fd-input-group__button--step-up sap-icon--slim-arrow-up"
      onClick={[Function]}
    />
    <button
      aria-label="Step down"
      className="fd-input-group__button fd-input-group__button--step-down sap-icon--slim-arrow-down"
      onClick={[Function]}
    />
  </span>
</div>
`;

exports[`<InputGroup /> create input group items 7`] = `
<div
  className="fd-input-group"
>
  <input
    className=""
    onChange={[Function]}
    placeholder="Search Term"
    type="search"
    value=""
  />
  <span
    className="fd-input-group__addon fd-input-group__addon--button"
  >
    <button
      aria-label="Clear"
      className="fd-input-group__button fd-input-group__button--clear"
      onClick={[Function]}
    />
  </span>
</div>
`;

exports[`<InputGroup /> create input group items 8`] = `
<div
  className="fd-input-group fd-input-group--compact"
>
  <input
    className="fd-input fd-input--compact"
    onChange={[Function]}
    placeholder="Search Term"
    type="search"
    value="search me"
  />
  <span
    className="fd-input-group__addon fd-input-group__addon--button"
  >
    <button
      aria-label="Clear"
      className="fd-input-group__button fd-input-group__button--clear"
      onClick={[Function]}
    />
  </span>
</div>
`;

exports[`<InputGroup /> create input group items 9`] = `
<div
  className="fd-input-group fd-input-group--before"
>
  <span
    className="fd-input-group__addon fd-input-group__addon--before"
  >
    <span
      className="sap-icon--globe"
      role="presentation"
    />
  </span>
  <input
    className=""
    onChange={[Function]}
    type="text"
    value="1234567890"
  />
</div>
`;

exports[`<InputGroup /> create input group items 10`] = `
<div
  className="fd-input-group fd-input-group--before fd-input-group--compact"
>
  <span
    className="fd-input-group__addon fd-input-group__addon--before"
  >
    <span
      className="sap-icon--globe"
      role="presentation"
    />
  </span>
  <input
    className="fd-input fd-input--compact"
    onChange={[Function]}
    type="text"
    value="1234567890"
  />
</div>
`;

exports[`<InputGroup /> create input group items 11`] = `
<div
  className="fd-input-group fd-input-group--after"
>
  <input
    className=""
    onChange={[Function]}
    type="text"
    value="1234567890"
  />
  <span
    className="fd-input-group__addon fd-input-group__addon--after"
  >
    <span
      className="sap-icon--hide"
      role="presentation"
    />
  </span>
</div>
`;

exports[`<InputGroup /> create input group items 12`] = `
<div
  className="fd-input-group fd-input-group--after fd-input-group--compact"
>
  <input
    className="fd-input fd-input--compact"
    onChange={[Function]}
    type="text"
    value="1234567890"
  />
  <span
    className="fd-input-group__addon fd-input-group__addon--after"
  >
    <span
      className="sap-icon--hide"
      role="presentation"
    />
  </span>
</div>
`;

exports[`<InputGroup /> create input group items 13`] = `
<div
  className="fd-input-group fd-input-group--after"
>
  <input
    className=""
    onChange={[Function]}
    type="text"
    value="1234567890"
  />
  <span
    className="fd-input-group__addon fd-input-group__addon--button fd-input-group__addon--after"
  >
    <button
      className="fd-button--light"
      disabled={false}
      selected={false}
    >
      Button
    </button>
  </span>
</div>
`;

exports[`<InputGroup /> create input group items 14`] = `
<div
  className="fd-input-group fd-input-group--after fd-input-group--compact"
>
  <input
    className="fd-input fd-input--compact"
    onChange={[Function]}
    type="text"
    value="1234567890"
  />
  <span
    className="fd-input-group__addon fd-input-group__addon--button fd-input-group__addon--after"
  >
    <button
      className="fd-button--light"
      disabled={false}
      selected={false}
    >
      Button
    </button>
  </span>
</div>
`;

exports[`<InputGroup /> create input group items 15`] = `
<div
  className="fd-input-group fd-input-group--before"
>
  <span
    className="fd-input-group__addon fd-input-group__addon--button fd-input-group__addon--before"
  />
  <input
    className=""
    onChange={[Function]}
    type="text"
    value="1234567890"
  />
</div>
`;

exports[`<InputGroup /> create input group items 16`] = `
<div
  className="fd-form__group"
>
  <div
    className="fd-input-group fd-input-group--after"
  >
    <input
      className=""
      onChange={[Function]}
      type="text"
      value="1234567890"
    />
    <span
      className="fd-input-group__addon fd-input-group__addon--after"
    >
      €
    </span>
  </div>
</div>
`;

exports[`<InputGroup /> create input group items 17`] = `"<InputGroup />"`;
